Sources/SourceKitLSP/DocumentManager.swift

```diff
@@ -15,16 +15,46 @@ import LanguageServerProtocol
 import LSPLogging
 import SKSupport
 
+public struct DocumentTokens {
+  /// Lexical tokens, e.g. keywords, raw identifiers, ...
+  public var lexical: [SemanticToken] = []
```

Comment:
I thought about this one too. LSP calls all tokens 'semantic tokens' and to me `SemanticToken` feels like a structure that directly reflects the LSP representation. But I can see your point, perhaps it would be better to rename it, since we internally have different 'kinds' of tokens.

Reply:
I would argue that if the type was defined in the `LanguageServerProtocol` module, we should stick to the LSP terminology. But this is more of an internal data structure IMO and we only “upgrade” lexical and syntactic tokens to semantic tokens on the LSP boundary. That’s my opinion anyway. I can see your point as well.